src/components/Product/Product.jsx

```diff
@@ -1,33 +1,169 @@
+
+function ProductModeShow({
+	id,
+	title,
+	detail,
+	email,
+	image,
+	altText,
+	onDeleteProduct,
+	onEnableEditMode,
+}) {
 	const handleDelete = () => {
 		onDeleteProduct(id);
 	};
-
 	return (
 		<Container>
 			<ImageWrapper>
-				<Image
-					src={image}
-					alt={altText}
-					width={3456}
-					height={5184}
-					layout="fill"
-					objectFit="cover"
-				/>
+				<Image src={image} alt={altText} layout="fill" objectFit="cover" />
 			</ImageWrapper>
 			<TextWrapper>
 				<Typography variant="p">{title}</Typography>
 				<Typography variant="p">{detail}</Typography>
-				<DefaultButton type="submit">{email}</DefaultButton>
+				<DefaultButton>{email}</DefaultButton>
 				<DefaultButton onClick={handleDelete}>delete</DefaultButton>
+				<DefaultButton onClick={onEnableEditMode}>edit</DefaultButton>
 			</TextWrapper>
 		</Container>
 	);
 }
+
+function ProductModeEdit({ id, title, detail, email, image, onDisableEditMode, onUpdateProduct }) {
+	const {
+		setValue,
+		watch,
+		register,
+		handleSubmit,
+		formState: { errors },
+	} = useForm();
+
+	const {
+		image: imageValue,
+		detail: detailValue,
+		email: emailValue,
+		title: titleValue,
+	} = watch();
+
+	useEffect(() => {
+		setValue('title', title);
+		setValue('detail', detail);
+		setValue('image', image);
+		setValue('email', email);
+	}, []);
+
+	const onSubmit = () => {
+		onDisableEditMode();
+	};
+
+	return (
+		<FormContainer>
+			<FormElement onSubmit={handleSubmit(onSubmit)}>
+				<InputContainer>
+					<InputSingleContainer>
+						<Label htmlFor="image">image url</Label>
+						<Input
+							id="image"
+							type="url"
+							aria-invalid={errors.image ? 'true' : 'false'}
+							{...register('image', {
+								required: 'Please fill in a url',
+							})}
+						/>
+					</InputSingleContainer>
+					<InputSingleContainer>
+						<Label htmlFor="title">title</Label>
+						<Input
+							id="title"
+							type="text"
+							aria-invalid={errors.title ? 'true' : 'false'}
+							{...register('title', {
+								required: 'short descriptive title',
+								maxLength: 20,
+							})}
+						/>
+					</InputSingleContainer>
+					<InputSingleContainer>
+						<Label htmlFor="detail">detail</Label>
+						<Input
+							id="detail"
+							type="text"
+							aria-invalid={errors.detail ? 'true' : 'false'}
+							{...register('detail', {
+								required: 'Add important details',
+								maxLength: 170,
+							})}
+						/>
+					</InputSingleContainer>
+					<InputSingleContainer>
+						<Label htmlFor="email">email</Label>
+						<Input
+							id="email"
+							type="email"
+							aria-invalid={errors.email ? 'true' : 'false'}
+							{...register('email', {
+								required: 'please add your email',
+								maxLength: 60,
+							})}
+						/>
+					</InputSingleContainer>
+				</InputContainer>
+				<ButtonContainer>
+					<DefaultButton
+						type="submit"
+						onClick={event => {
+							event.preventDefault();
+							onUpdateProduct(
```

Comment:
Just curious: what does the "on.." in your wording stand for?

Reply:
this is how we learned it. This is about state uplifting. In the index.js I'm creating an updateProduct function and I'm adding an onUpdateProduct={updateProduct} attribute to my productGrid